### ISSUE TITLE
Punch copy and Punch paste as a function and not as a plugins

### ISSUE THREAD

**Issue by Reaper10** [tenacityteam/tenacity-legacy#370]
I think the Punch copy and paste plugin that can be found here 
https://forum.audacityteam.org/viewtopic.php?t=71556
It should be standard editing functions in the editing toolbar.

![Screenshot 2021-07-21 202310](https://user-images.githubusercontent.com/8137561/126880986-865dbc3a-9945-4909-b8cd-cffe1cd0f30d.jpg)

That being said I have opened Punch copy and Punch paste function icon discussion #371  to discussion  the icon design. If these the punch copy and punch paste plugins are made into function you’ll need this automation #344. Why did I do this, you may ask. It’ll speed up the work flow by eliminating the need for these options  

![Screenshot 2021-07-24 145956](https://user-images.githubusercontent.com/8137561/126881013-e24e8e51-deff-4ca1-b689-744f544981c0.jpg)

not to mention the need for a preview button and cross fade types because a user can do this with an automation track. This function would be real time a user could see and or edit the cross fade right on the track in question.

Possible hot keys for this

Punch copy crrl shift c

Punch paste crrl shift v
 [ x] I have read the specified guidelines for issues
@akleja has a better idea
 There might be multiple ways to implement something similar to this. The current volume envelope could be utilized.

The behaviour I would expect is when one clip is dragged over another (non destructive) volume envelopes should automatically be applied. The length of the crossfade should be determined by the length of the overlap
And the intersecting regions (middle line, clip A end, clip B start) should be possible to slide left/right.
When pasting over a pre-existing item it should replace parts of the previous item and apply the same type of envelope-based cross fades. For this a default length of crossfades needs to be specified (and possible to change).

**Comment by akleja** [comment]
There might be multiple ways to implement something similar to this. The current volume envelope could be utilized.

The behaviour I would expect is when one clip is dragged over another (non destructive) volume envelopes should automatically be applied. The length of the crossfade should be determined by the length of the overlap
And the intersecting regions (middle line, clip A end, clip B start) should be possible to slide left/right.
When pasting over a pre-existing item it should replace parts of the previous item and apply the same type of envelope-based cross fades. For this a default length of crossfades needs to be specified (and possible to change).

That's just my thoughts

**Comment by Reaper10** [comment]
> There might be multiple ways to implement something similar to this. The current volume envelope could be utilized.
> 
> The behaviour I would expect is when one clip is dragged over another (non destructive) volume envelopes should automatically be applied. The length of the crossfade should be determined by the length of the overlap
> And the intersecting regions (middle line, clip A end, clip B start) should be possible to slide left/right.
> When pasting over a pre-existing item it should replace parts of the previous item and apply the same type of envelope-based cross fades. For this a default length of crossfades needs to be specified (and possible to change).
> 
> That's just my thoughts

I like this better just need a ui for it.

**Comment by akleja** [comment]
Yeah the way I envision it It wouldn't really need many UI changes at all, but rather mostly work with what's already there.
